### PR TITLE
Unify the time unit for intervals to microseconds.

### PR DESCRIPTION
**Pull request by malin1993ml**
We sometimes use microseconds and sometimes use milliseconds to set the intervals for the components that wake up periodically. This can lead to confusion and sometimes we even have the wrong specification in settings (e.g., we use `us` for `LogSerializerTask` for in settings it says `ms`). 

This PR unifies all these intervals to use `us`. And this also gives us more flexibility to set finer-grained intervals.

In addition, this PR:
1. Cap exponential back-off logic for `LogSerializerTask` to 10000us.
2. Set the intervals to some values I think "reasonable":
LogSerializerTask: 100us
DiskLogConsumerTask: 100us
GarbageCollectorThread: 1000us
MetricsThread: 10000us
3. Fix typos.

### CODE REVIEW

**Comment by mbutrovich** [comment]
Do we know what the performance implications are for these settings? For example changing the log serialization interval means that the latch that committing transactions need will be grabbed by the serialization thread every 100 microseconds. That seems aggressive but don't have numbers to back it up.

**Comment by malin1993ml** [comment]
We could wait to see the benchmarking results from the logging group. And if 100us is too aggressive, we could just bump them up to 1000us. Regardless, I think unifying everything to us is helpful.

File: benchmark/integration/tpcc_benchmark.cpp [hunk 1]
```diff
@@ -41,8 +41,8 @@ class TPCCBenchmark : public benchmark::Fixture {
 
   // Settings for log manager
   const uint64_t num_log_buffers_ = 100;
-  const std::chrono::microseconds log_serialization_interval_{5};
-  const std::chrono::milliseconds log_persist_interval_{10};
+  const std::chrono::microseconds log_serialization_interval_{100};
```
[the remark this answers is not in the document]
Comment (eppingere) [reply benchmark/integration/tpcc_benchmark.cpp:44]: Also just across the board, except ^, it seems like you have reduced the wait times by a factor of 10 or 100. I guess I'm wondering why this one is increased?

[the remark this answers is not in the document]
Comment (malin1993ml) [reply benchmark/integration/tpcc_benchmark.cpp:44]: It is indeed an increase, but from 5**us** to 100**us**. In general, I just set the interval of `LogSerializerTask` to be 100us across the board. Of cause, it's not a super fine-tuned value, but this interval is small enough compare to the disk latency (~several milliseconds), and bigger than the original value used in the system (5-10us), which can reduce the latch contention Matt mentioned as well. It's good to measure the performance and double-check though.

[the remark this answers is not in the document]
Comment (eppingere) [reply benchmark/integration/tpcc_benchmark.cpp:44]: Ok thanks that makes sense

**Comment by malin1993ml** [comment]
@thepinetree And what was the idle CPU usage before this PR?

**Comment by eppingere** [comment]
> @thepinetree And what was the idle CPU usage before this PR?

0.5-3%. Its essentially the same on this branch once it hits max sleep.

**Pull request merged**